### DISCUSSION TITLE
Wyze Lock Bolt Detection in Home Assistant

### DISCUSSION THREAD

**Discussion by SDeGonge** [SecKatie/ha-wyzeapi#522] [Q&A]
I've set up HACS & the Wyze integration according to the instructions.
I just installed my Wyze Lock Bolt and set ut up in the Wyze app.
Restarted HomeAssistant and expecte to see my lock listed in the integration.
Instead, I only see the following...
  Wyze Notifications
  Front Door.keypad_battery (which is reporting as unavailable)

Any idea why I'm not seeing the "lock." ?
I was under the impression that the lock was supported by this integration.

Thanks

**Comment by DubMFG** [comment]
The Wyze Lock (not the Wyze Lock **Bolt**) is available via this integration. The Lock Bolt does not support proximity features (at least natively from wyze - there is the ability to use a camera to unlock from the app, so I PRESUME there's an API call that hasn't been weaponized yet)

**Comment by Resinchem** [comment]
Just FYI in case anyone runs into the same issue... I have multiple Wyze cams and a lock/keypad.  All the cameras integrate just fine (RTSP) with Home Assistant, but the lock disappeared a while back and I had been unable to get it to reappear into Home Assistant.  After much reading, research, reloading the integration, testing, etc., I found that it was actually PiHole that was blocking the API.  If anyone else runs into these issues, I found that I needed to whitelist the following domains:

api.wyzecam.com
yd-saas-toc.wyzecam.com

Once I whitelisted these in PiHole and restarted Home Assistant, the lock (and battery) reappeared in Home Assistant.  Maybe this will help out someone else that can't get the lock integrated via the Wyze integration.  If that doesn't work and you are running some sort of ad-blocker, try temporarily disabling the ad-blocker and reload the integration and/or restart Home Assistant.  If the lock appears, then you've located the culprit!

**Comment by dbortz** [comment]
For anyone landing here trying to get the Lock Bolt actually working in HA: I wrote up a full step-by-step — `ha-wyzeapi` + an ESPHome Bluetooth proxy (local BLE lock/unlock, with all the gotchas like the "invalid credentials" cred-order trap and needing an *active* proxy) → https://github.com/SecKatie/ha-wyzeapi/discussions/866

**Comment by tpawley2001** [comment]
FYI for anyone who gets the Bolt working over BLE from this thread — the **battery** still comes only from the Wyze **cloud** (`power` field), which on a gateway-less Bolt is frequently stale or `unavailable` (it only refreshes when a phone syncs the lock).

The lock actually reports **live** battery over BLE too. The `0x2A19` Battery Level characteristic returns 16 bytes of AES-128-ECB ciphertext, encrypted with the **same key ha-wyzeapi already derives for lock state** (`uuid[-16:].lower()`). Decrypted it's the same layout as the state characteristic, and `byte[0]` is the current percentage (with its own timestamp, so it's fresher than the cloud value).

I put together a small coordinator patch + ESP32 active-proxy config + a standalone decode demo here: https://github.com/tpawley2001/Wyze-Bolt-WiFi-Bridge-ESP32 (more detail in #833). Complements the BLE lock/unlock setup already described in this thread.
